File: strategy/typescript/glados/agent/shared/halt.lua
local Base = require "agent/base/behavior"
local Halt = Class("Agent.Shared.Halt", Base)

local World = require "../base/world"
local HaltTask = require "task/shared/halt"


function Halt:check()
	return World.RefereeState == "Halt"
end

function Halt:_updateTask()
	return HaltTask
end

return Halt

File: strategy/typescript/glados/agent/shared/halt.ts
import * as World from "base/world";

import { Behavior, TaskAssignment } from "glados/agent/base/behavior";
import { Halt as HaltTask } from "glados/task/shared/halt";

export class Halt extends Behavior {
	check(): boolean {
		return World.RefereeState === "Halt";
	}

	_updateTask(): TaskAssignment<typeof HaltTask> {
		return [HaltTask];
	}
}
